server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/method/EmailSubmissionSetMethod.scala

```diff
@@ -199,30 +199,38 @@ class EmailSubmissionSetMethod @Inject()(serializer: EmailSubmissionSetSerialize
   private def create(request: EmailSubmissionSetRequest,
                      session: MailboxSession,
                      processingContext: ProcessingContext): SMono[(CreationResults, ProcessingContext)] =
-    SFlux.fromIterable(request.create
-      .getOrElse(Map.empty)
-      .view)
-      .fold((CreationResults(Nil), processingContext)) {
-        (acc : (CreationResults, ProcessingContext), elem: (EmailSubmissionCreationId, JsObject)) => {
+    SFlux.fromIterable(
+      request.create
+        .getOrElse(Map.empty)
+        .view)
+      .fold[SMono[(CreationResults, ProcessingContext)]](SMono.just((CreationResults(Nil), processingContext))) {
+        (acc: SMono[(CreationResults, ProcessingContext)], elem: (EmailSubmissionCreationId, JsObject)) => {
           val (emailSubmissionCreationId, jsObject) = elem
-          val (creationResult, updatedProcessingContext) = createSubmission(session, emailSubmissionCreationId, jsObject, acc._2)
-          (CreationResults(acc._1.created :+ creationResult), updatedProcessingContext)
+          acc.flatMap {
+            case (creationResults, processingContext) =>
+              createSubmission(session, emailSubmissionCreationId, jsObject, processingContext)
+                .map {
+                  case (created, updatedProcessingContext) => CreationResults(creationResults.created :+ created) -> updatedProcessingContext
+                }
+              .switchIfEmpty(SMono.error(new RuntimeException("I should not be empty")))
+          }.cache()
         }
       }
+      .flatMap(x => x)
```

Comment:
x?

Reply:
Yes `_` misbehaved in my IDE, this is clearly an identity function so I don't think the name matter.